src/Nix/Builtins.hs

```diff
@@ -705,39 +721,44 @@ genericClosure = fromValue @(AttrSet (NThunk m)) >=> \s ->
 
 replaceStrings :: MonadNix e m => m (NValue m) -> m (NValue m) -> m (NValue m) -> m (NValue m)
```

Comment:
Oof this one is complicated. We should come back to it and document the logic once all of the context stuff is in place.